web/ui/module/codemirror-promql/src/parser/matcher.ts

```diff
@@ -23,22 +35,57 @@ function createMatcher(labelMatcher: SyntaxNode, state: EditorState): Matcher {
     // weird case, that would mean the labelMatcher doesn't have any child.
     return matcher;
   }
-  do {
-    switch (cursor.type.id) {
-      case LabelName:
-        matcher.name = state.sliceDoc(cursor.from, cursor.to);
-        break;
-      case MatchOp:
-        const ope = cursor.node.firstChild;
-        if (ope) {
-          matcher.type = ope.type.id;
+  switch (cursor.type.id) {
+    case QuotedLabelMatcher:
+      if (!cursor.next()) {
```

Comment:
since you already check that line 34, you don't need to do it again

Reply:
I think this is required as we need to move to the next node in the tree. 
Also my bad, I have written wrong comments. It should be `// weird case, that would mean the QuotedLabelMatcher doesn't have any child.`

Reply:
I've updates the comments.

Reply:
ah yeah you are right with the current grammar format, it's required my bad.
With the simplification I have suggested, that wouldn't be required